csci-499-fa24/AccessAbleNYC
# Client/map component/route/instruction summary

## Description
Updated RoutingMachine component to improved the instructions box.  Improved logic for instruction display and UI. Instructions box now matches the theme (light or dark) and allows users to click on individual steps to focus on specific map points. Clicking an instruction highlights the point and centers the map on it for better navigation.


## What's in this change?
-  Improved Instructions Box: Improved logic for instruction display, enhanced box styling, and added custom scrollbars compatible with macOS.
- Added clickable Instructions step with hovering effect 
- Added dark mode for Instructions box 

## Testing changes

- Clickable Instructions: Verified that clicking an instruction step highlights the point and centers the map correctly.
- Hover Effects: Confirmed that hover feedback works as expected in both themes.
- Scroll Behavior: Checked smooth scrolling and custom scrollbar styling.
- Theme Switching: Tested light and dark modes to confirm correct styling of the instructions box.
